### PR TITLE
Initialization states

### PR DESCRIPTION
**Pull request by scotthasbrouck**

### CODE REVIEW

**Comment by scotthasbrouck** [comment]
@dmitrig01 feedback on this when you get a chance?
Also, let me know if you want me to keep looking into including the app_id in the call to referrals.

File: src/3_branch.js [hunk 1]
```diff
@@ -343,10 +369,11 @@ if (CORDOVA_BUILD) { // jshint undef:false
...
  */
 Branch.prototype['setIdentity'] = wrap(callback_params.CALLBACK_ERR_DATA, function(done, identity) {
+	var self = this;
 	this._api(resources.profile, { "identity": identity }, function(err, data) {
-		this.identity_id = data['identity_id'];
-		this.sessionLink = data['link'];
-		this.identity = data['identity'];
+		self.identity_id = data['identity_id'];
+		self.sessionLink = data['link'];
+		self.identity = data['identity'];
```
Comment (dmitrig01) [inline src/3_branch.js]: ack, great catch

**Comment by scotthasbrouck** [comment]
Haha, yeah.
Good to merge this?

**Comment by dmitrig01** [comment]
i've now gotten halfway through writing this comment (and then getting distracted) at least twice, but:

it looks great! the one change i might make is - not sure if this belongs in the docs. for one, we don't actually expose init_state, since the minified will rename the variable to a different name. but beyond that, what's the use case for someone actually using it?

**Comment by scotthasbrouck** [comment]
GREAT point. haha...
One thing that comes to mind is rather than just getting an error returned by init, perhaps they would want to stop making further SDK requests if init_state is failed? It would be easy to expose, but perhaps it's not worth adding to the amount of information already in the Readme, it is getting rather long. I'm fine not mentioning it in the docs.

**Comment by dmitrig01** [comment]
yeah, i think as long as the error is easy to catch it's fine

maybe we could say something like "if init fails, all subsequent calls will also fail"

**Comment by scotthasbrouck** [comment]
That makes sense. I'll make those changes now

**Comment by scotthasbrouck** [comment]
Ok, should be all set

**Comment by dmitrig01** [comment]
:+1: good work

**Pull request merged**